tree/ntupleutil/v7/src/RNTupleInspector.cxx

```diff
@@ -61,8 +61,13 @@ void ROOT::Experimental::RNTupleInspector::CollectColumnInfo()
 
          if (fCompressionSettings == -1) {
             fCompressionSettings = columnRange.fCompressionSettings;
-         } else {
-            R__ASSERT(columnRange.fCompressionSettings == fCompressionSettings);
+         } else if (fCompressionSettings != columnRange.fCompressionSettings) {
+            // Note that currently all clusters and columns are compressed with the same settings and it is not yet
+            // possible to do otherwise. This measn that currently, this exception should never be thrown, but this
+            // could change in the future.
+            throw RException(R__FAIL("compression setting mismatch between column ranges (" +
+                                     std::to_string(fCompressionSettings) + " vs " +
+                                     std::to_string(columnRange.fCompressionSettings) + ")"));
```

Comment:
Actually, IMHO, it might make more sense to not throw here, and only only update `fCompressionSettings` if compression stays the same throughout all columns in all clusters.
Thus, `GetCompressionSettings()` may either return a valid compression algorithm + level, or a special value signalling that it varies among columns.  Any take on this @vepadulano, @jblomer?

Reply:
Can a `RNTupleInspector` be used to look at a `RNtuple` being written to?  If yes, do we have (a need for) an interface to return the compression setting that will be used on to the next page (independently of the compression that has been used so far)?  [I also assume we have an accessor for the 'overall' compression ration]

Another alternative would also to return a 'collection' of compression settings in case there is more than one.

Reply:
I'm not fully sure if I understand the question correctly, do you mean using the `RNTupleInspector` as a kind of monitoring tool while writing? This is currently not the intended use, but might be useful (although perhaps under a different name, to not muddle the purpose of `RNTupleInspector`, which is for statically analysing existing `RNTuples`).

There currently is no accessor for the overall compression, hence this current approach. There also is no way yet to have different compression across different pages, but once that becomes possible we should rethink this function altogether.